samples/csharp_dotnetcore/51.cafe-bot/Dialogs/Dispatcher/InterruptionDispatcher.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Dialogs;
+
+namespace Microsoft.BotBuilderSamples
+{
+    public class InterruptionDispatcher : ComponentDialog
+    {
+        private const string NoneIntent = "None";
+        private const string InterruptionDispatchDialog = "interruptionDispatcherDialog";
+
+        private IStatePropertyAccessor<OnTurnProperty> _onTurnAccessor;
+        private ConversationState _conversationState;
+        private IStatePropertyAccessor<UserProfile> _userProfileAccessor;
+        private BotServices _botServices;
+
+        /**
+         * Constructor.
+         *
+         * @param {StatePropertyAccessor} onTurnAccessor
+         * @param {ConversationState} conversationState
+         * @param {StatePropertyAccessor} userProfileAccessor
+         * @param {BotConfiguration} botConfig
+         */
```

Comment:
This file contains javascript style comments, consider the c# xml style comments